Docker.

Move code to the Docker container.

=> ERROR [4/4] RUN g++ -o FamilyBudgetCountLib main.cpp                                                                                               1.7s
------
 > [4/4] RUN g++ -o FamilyBudgetCountLib main.cpp:
1.549 main.cpp:6:10: fatal error: crow.h: No such file or directory
1.549     6 | #include <crow.h>
1.549       |          ^~~~~~~~
1.549 compilation terminated.
------
Dockerfile:11
--------------------
   9 |
  10 |     # Compile the C++ program
  11 | >>> RUN g++ -o FamilyBudgetCountLib main.cpp
  12 |
  13 |     # Specify the command to run the executable
--------------------
ERROR: failed to solve: process "/bin/sh -c g++ -o FamilyBudgetCountLib main.cpp" did not complete successfully: exit code: 1

D:\Applications\FBCountLib\FamilyBudgetCountLib>

> [4/4] RUN g++ -o FamilyBudgetCountLib main.cpp:
3.986 main.cpp:6:10: fatal error: crow.h: No such file or directory
3.986     6 | #include <crow.h>
3.986       |          ^~~~~~~~
3.986 compilation terminated.
------
Dockerfile:11

> [5/5] RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp:
1.142 In file included from /usr/src/crow/include/crow.h:7,
1.142                  from main.cpp:6:
1.142 /usr/src/crow/include/crow/socket_adaptors.h:13:10: fatal error: asio.hpp: No such file or directory
1.142    13 | #include <asio.hpp>
1.142       |          ^~~~~~~~~~
1.142 compilation terminated.

=> ERROR [6/6] RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp                                                                      1.0s
------
 > [6/6] RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp:
0.918 In file included from /usr/src/crow/include/crow.h:7,
0.918                  from main.cpp:6:
0.918 /usr/src/crow/include/crow/socket_adaptors.h:13:10: fatal error: asio.hpp: No such file or directory
0.918    13 | #include <asio.hpp>
0.918       |          ^~~~~~~~~~
0.918 compilation terminated.

> [6/6] RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp -lboost_system -lboost_thread -lboost_filesystem -DCROW_USE_BOOST:
18.76 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#1}::operator()(double) const':
18.77 main.cpp:(.text+0x5fa): undefined reference to `getIncomeGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#2}::operator()(double) const':
18.77 main.cpp:(.text+0x6e0): undefined reference to `getMyGroceriesGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#3}::operator()(double) const':
18.77 main.cpp:(.text+0x7c6): undefined reference to `getMyHousingGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#4}::operator()(double) const':
18.77 main.cpp:(.text+0x8ac): undefined reference to `getMyTravellingGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#5}::operator()(double) const':
18.77 main.cpp:(.text+0x992): undefined reference to `getMyEntertainmentGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#6}::operator()(double) const':
18.77 main.cpp:(.text+0xa78): undefined reference to `getMyHealthGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `main::{lambda(double)#7}::operator()(double) const':
18.77 main.cpp:(.text+0xb5e): undefined reference to `getMyMobileInternetGraphics(double)'
18.77 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetIncomeGraphics()':
18.78 main.cpp:(.text+0x13ed): undefined reference to `getIncomeGraphics(double)'
18.78 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyGroceriesGraphics()':
18.78 main.cpp:(.text+0x14d3): undefined reference to `getMyGroceriesGraphics(double)'
18.78 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyHousingGraphics()':
18.78 main.cpp:(.text+0x15b9): undefined reference to `getMyHousingGraphics(double)'
18.78 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyTravellingGraphics()':
18.78 main.cpp:(.text+0x169f): undefined reference to `getMyTravellingGraphics(double)'
18.78 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyEntertainmentGraphics()':
18.79 main.cpp:(.text+0x1785): undefined reference to `getMyEntertainmentGraphics(double)'
18.79 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyHealthGraphics()':
18.79 main.cpp:(.text+0x186b): undefined reference to `getMyHealthGraphics(double)'
18.79 /usr/bin/ld: /tmp/ccBA8YCl.o: in function `testGetMyMobileInternetGraphics()':
18.79 main.cpp:(.text+0x1951): undefined reference to `getMyMobileInternetGraphics(double)'
18.84 collect2: error: ld returned 1 exit status

--- a/Dockerfile
+++ b/Dockerfile
@@ -8,7 +8,7 @@ WORKDIR /usr/src/app
 COPY . .
 
 # Crow library
-RUN apt-get update && apt-get install -y git cmake libboost-all-dev
+# RUN apt-get update && apt-get install -y git cmake libboost-all-dev
 RUN git clone https://github.com/CrowCpp/crow.git /usr/src/crow
 
 # Compile the C++ program

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,7 +11,7 @@ COPY . .
 # Crow library
 RUN git clone https://github.com/CrowCpp/crow.git /usr/src/crow
 # ASION lib
-RUN apt-get install -y asio-dev
+RUN apt-get install -y libboost-all-dev
 
 # Compile the C++ program
 RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,8 +7,12 @@ WORKDIR /usr/src/app
 # Copy the current directory contents into the container
 COPY . .
 
+# Crow library
+RUN apt-get update && apt-get install -y git cmake libboost-all-dev
+RUN git clone https://github.com/CrowCpp/crow.git /usr/src/crow
+
 # Compile the C++ program
-RUN g++ -o FamilyBudgetCountLib main.cpp
+RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -10,8 +10,13 @@ COPY . .
 # RUN apt-get update && apt-get install -y git cmake libboost-all-dev
 # Crow library
 RUN git clone https://github.com/CrowCpp/crow.git /usr/src/crow
-# ASION lib
-RUN apt-get install -y libboost-all-dev
+RUN apt-get update && apt-get install -y \
+    git \
+    cmake \
+    libboost-system-dev \
+    libboost-thread-dev \
+    libboost-filesystem-dev \
+    libboost-asio-dev
 
 # Compile the C++ program
 RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp

--- a/Dockerfile
+++ b/Dockerfile
@@ -8,7 +8,7 @@ WORKDIR /usr/src/app
 COPY . .
 
 # Compile the C++ program
-RUN g++ -o main.cpp
+RUN g++ -o  FamilyBudgetCountLib main.cpp
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -15,8 +15,8 @@ RUN apt-get update && apt-get install -y \
     cmake \
     libboost-system-dev \
     libboost-thread-dev \
-    libboost-filesystem-dev \
-    libboost-asio-dev
+    libboost-filesystem-dev 
+    
 
 # Compile the C++ program
 RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp

--- a/Dockerfile
+++ b/Dockerfile
@@ -15,11 +15,13 @@ RUN apt-get update && apt-get install -y \
     cmake \
     libboost-system-dev \
     libboost-thread-dev \
-    libboost-filesystem-dev 
+    libboost-filesystem-dev \
+    libboost-dev
     
 
 # Compile the C++ program
-RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp
+#RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp
+RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp -lboost_system -lboost_thread -lboost_filesystem
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -21,7 +21,7 @@ RUN apt-get update && apt-get install -y \
 
 # Compile the C++ program
 #RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp
-RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp -lboost_system -lboost_thread -lboost_filesystem -DCROW_USE_BOOST
+RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp functionslib.cpp -lboost_system -lboost_thread -lboost_filesystem -DCROW_USE_BOOST
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,9 +7,11 @@ WORKDIR /usr/src/app
 # Copy the current directory contents into the container
 COPY . .
 
-# Crow library
 # RUN apt-get update && apt-get install -y git cmake libboost-all-dev
+# Crow library
 RUN git clone https://github.com/CrowCpp/crow.git /usr/src/crow
+# ASION lib
+RUN apt-get install -y asio-dev
 
 # Compile the C++ program
 RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,14 @@
+# Use an official GCC image from Docker Hub
+FROM gcc:latest
+
+# Set the working directory inside the container
+WORKDIR /usr/src/app
+
+# Copy the current directory contents into the container
+COPY . .
+
+# Compile the C++ program
+RUN g++ -o FamilyBudgetCountLib main.cpp
+
+# Specify the command to run the executable
+CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -8,7 +8,7 @@ WORKDIR /usr/src/app
 COPY . .
 
 # Compile the C++ program
-RUN g++ -o  FamilyBudgetCountLib main.cpp
+RUN g++ -o FamilyBudgetCountLib main.cpp
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -21,7 +21,7 @@ RUN apt-get update && apt-get install -y \
 
 # Compile the C++ program
 #RUN g++ -I /usr/src/crow/include -o FamilyBudgetCountLib main.cpp
-RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp -lboost_system -lboost_thread -lboost_filesystem
+RUN g++ -I/usr/src/crow/include -I/usr/include/boost -o FamilyBudgetCountLib main.cpp -lboost_system -lboost_thread -lboost_filesystem -DCROW_USE_BOOST
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -8,7 +8,7 @@ WORKDIR /usr/src/app
 COPY . .
 
 # Compile the C++ program
-RUN g++ -o FamilyBudgetCountLib main.cpp
+RUN g++ -o main.cpp
 
 # Specify the command to run the executable
 CMD ["./FamilyBudgetCountLib"]

--- a/FamilyBudgetCountLib.vcxproj
+++ b/FamilyBudgetCountLib.vcxproj
@@ -141,6 +141,7 @@
     <Text Include="CMakeLists.txt" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="Dockerfile" />
     <None Include="Makefile" />
     <None Include="Procfile" />
     <None Include="testEndpoints.ps1" />

--- a/FamilyBudgetCountLib.vcxproj.filters
+++ b/FamilyBudgetCountLib.vcxproj.filters
@@ -35,5 +35,6 @@
     <None Include="testEndpoints.ps1" />
     <None Include="Makefile" />
     <None Include="Procfile" />
+    <None Include="Dockerfile" />
   </ItemGroup>
 </Project>

